url49584c1d41_Arrem_GGS_tgz-pJ8-com_gamezgalaxy_test_console_MainJ8/src/com/gamezgalaxy/GGS/world/convert/DatToGGS.java

```diff
@@ -0,0 +1,181 @@
+/*******************************************************************************
+ * Copyright (c) 2012 GamezGalaxy.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Public License v3.0
+ * which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/gpl.html
+ ******************************************************************************/
+package com.gamezgalaxy.GGS.world.convert;
+
+import java.io.*;
+import java.text.DateFormat;
+import java.util.zip.GZIPOutputStream;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * This class was taken from the Minecraft Wiki
+ * and was slightly modified
+ *
+ */
+public class DatToGGS {
+	public com.mojang.minecraft.level.Level level = null;
+	private int width;
+	private int height;
+
+	public DatToGGS() {
+	}
+
+	public void load(String filename) {
+		FileInputStream fis = null;
+		GZIPInputStream gzis = null;
+		ObjectInputStream in = null;
+		DataInputStream inputstream = null;
+		try {
+			fis = new FileInputStream(filename);
+			gzis = new GZIPInputStream(fis);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *gzis*. The resource is referenced at lines 35, 36, 37, 43, 50, 51. The resource is closed at lines 38, 39, 40, 45, 46, 47, 52, 53, 55, 56. There are other execution paths that do not contain closure statements, for example, when ObjectInputStream.readObject throws an exception, and an else condition at lines 37, 43 evaluate to true. Ensure *gzis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).